### PR TITLE
Resolving issue #378- Check ValidationUtils for negative Susceptance

### PR DESCRIPTION
**Pull request by Utkarsha-dev05**
Adapting ValidationUtils and test data such that only negative values are allowed for the transformer susceptance.

### CODE REVIEW

File: src/main/java/edu/ie3/datamodel/utils/validation/ConnectorValidationUtils.java [hunk 1]
```diff
@@ -80,7 +80,7 @@ private static void checkLine(LineInput line) {
   /**
    * Validates a line type if: <br>
    * - it is not null <br>
-   * - B is greater/equal to 0 (Phase-to-ground susceptance per length) <br>
+   * - B is less than 0 (Phase-to-ground susceptance per length) <br>
```
Comment (ckittl) [inline src/main/java/edu/ie3/datamodel/utils/validation/ConnectorValidationUtils.java:83]: Sorry for the flaw in issue description. It is fine, if the succeptance is zero. So negative values + zero shall pass.

File: src/main/java/edu/ie3/datamodel/utils/validation/ConnectorValidationUtils.java [hunk 2]
```diff
@@ -153,6 +153,8 @@ protected static void checkTransformer2WType(Transformer2WTypeInput transformer2
           transformer2WType.getxSc()
         },
         transformer2WType);
+    detectZeroOrPositiveQuantities(
+        new Quantity<?>[] {transformer2WType.getbM()}, transformer2WType);
```
Comment (ckittl) [inline src/main/java/edu/ie3/datamodel/utils/validation/ConnectorValidationUtils.java:157]: Please remove the check for negative quantity (cf. line 143)

**Review by ckittl — CHANGES_REQUESTED** [review]
Please additionally check the CI results. Currently some tests are broken.

File: src/main/java/edu/ie3/datamodel/utils/validation/ConnectorValidationUtils.java (hunk 1 above)
Comment (Utkarsha-dev05) [reply src/main/java/edu/ie3/datamodel/utils/validation/ConnectorValidationUtils.java:83]: Made the required changes

File: src/main/java/edu/ie3/datamodel/utils/validation/ConnectorValidationUtils.java (hunk 2 above)
Comment (Utkarsha-dev05) [reply src/main/java/edu/ie3/datamodel/utils/validation/ConnectorValidationUtils.java]: Done

File: src/main/java/edu/ie3/datamodel/utils/validation/ConnectorValidationUtils.java (hunk 1 above)
Comment (ckittl) [inline src/main/java/edu/ie3/datamodel/utils/validation/ConnectorValidationUtils.java:83]:
```suggestion
   * - B is less than / equal to 0 (Phase-to-ground susceptance per length) <br>
```

File: src/main/java/edu/ie3/datamodel/utils/validation/ConnectorValidationUtils.java [hunk 3]
```diff
@@ -124,7 +124,7 @@ private static void checkTransformer2W(Transformer2WInput transformer2W) {
    * - rSc is greater 0 (short circuit resistance) <br>
    * - xSc is greater 0 (short circuit impedance) <br>
    * - gM is greater/equal to 0 (no load conductance) <br>
-   * - bM is greater/equal to 0 (no load susceptance) <br>
+   * - bM is less/equal to 0 <br>
```
Comment (ckittl) [inline src/main/java/edu/ie3/datamodel/utils/validation/ConnectorValidationUtils.java:127]:
```suggestion
   * - bM is less/equal to 0 (no load susceptance)<br>
```

Please keep this part of the documentation

File: src/main/java/edu/ie3/datamodel/utils/validation/ConnectorValidationUtils.java [hunk 4]
```diff
@@ -193,7 +191,7 @@ private static void checkTransformer3W(Transformer3WInput transformer3W) {
    * - rScA, rScB, rScC are greater 0 (short circuit resistance in branches A,B,C) <br>
    * - xScA, xScB, xScC are greater 0 (short circuit impedance in branches A,B,C) <br>
    * - gM is greater/equal to 0 (no load conductance) <br>
-   * - bM is greater/equal to 0 (no load susceptance) <br>
+   * - bM is less/equal to 0 <br>
```
Comment (ckittl) [inline src/main/java/edu/ie3/datamodel/utils/validation/ConnectorValidationUtils.java:194]:
```suggestion
   * - bM is greater/equal to 0 (no load susceptance) <br>
```

File: src/main/java/edu/ie3/datamodel/utils/validation/ValidationUtils.java [hunk 5]
```diff
@@ -186,7 +186,18 @@ protected static void detectZeroOrNegativeQuantities(
     detectMalformedQuantities(
         quantities, entity, predicate, "The following quantities have to be positive");
   }
-
+  /**
+   * Goes through the provided quantities and reports those, that have zero or positive value via
+   * synoptic {@link UnsafeEntityException}
```
Comment (ckittl) [inline src/main/java/edu/ie3/datamodel/utils/validation/ValidationUtils.java:191]:
```suggestion
   * Goes through the provided quantities and reports those, that have positive value via
```

**Review by ckittl — CHANGES_REQUESTED** [review]
Generally looks good! Only some minor change requests regarding the documentation :-)

File: src/main/java/edu/ie3/datamodel/utils/validation/ConnectorValidationUtils.java (hunk 4 above)
Comment (Utkarsha-dev05) [reply src/main/java/edu/ie3/datamodel/utils/validation/ConnectorValidationUtils.java:194]: bM should be negative so shouldn't it be **less/equal to 0**?

Comment (ckittl) [reply src/main/java/edu/ie3/datamodel/utils/validation/ConnectorValidationUtils.java:194]: You are right! Sorry, my bad!

**Review by ckittl — APPROVED** [review]

**Pull request merged**